deploy/test/helm-props/k8s/helm-app.yaml

```diff
@@ -0,0 +1,29 @@
+# This app is created to run end-to-end with the Secrets Provider Job
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: test-env{{ ID }}
+  name: test-env{{ ID }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: test-env{{ ID }}
+  template:
+    metadata:
+      labels:
+        app: test-env{{ ID }}
+    spec:
+      serviceAccountName:  {{ SERVICE_ACCOUNT }}
+      containers:
+      - image: centos:7
+        name: test-app{{ ID }}
+        command: ["sleep"]
+        args: ["infinity"]
+        env:
+        - name: TEST_SECRET{{ ID }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ K8S_SECRETS }}
```

Comment:
is this the k8s secret name? if so, let's call it `K8S_SECRET_NAME`

Reply:
It is intended to be an array of secrets

Reply:
then this will not work. you are telling k8s to inject the value of the k8s secret into an env var but you are supplying it with an array of k8s secrets